CesiumGltf/include/CesiumGltf/MetadataConversions.h

```diff
@@ -210,64 +235,107 @@ struct MetadataConversions<
 };
 
 /**
- * @brief Converts from std::string_view to an unsigned integer.
+ * @brief Converts from std::string to an unsigned integer.
  */
 template <typename TTo>
 struct MetadataConversions<
     TTo,
-    std::string_view,
+    std::string,
     std::enable_if_t<
         CesiumGltf::IsMetadataInteger<TTo>::value && !std::is_signed_v<TTo>>> {
   /**
-   * @brief Converts the contents of a std::string_view to an signed integer.
-   * This assumes that the entire std::string_view represents the number, not
+   * @brief Converts the contents of a std::string to an signed integer.
+   * This assumes that the entire std::string represents the number, not
    * just a part of it.
    *
    * This returns std::nullopt if no number is parsed from the string.
    *
-   * @param from The std::string_view to parse from.
+   * @param from The std::string to parse from.
    * @param defaultValue The default value to be returned if conversion fails.
    */
-  static std::optional<TTo> convert(const std::string_view& from) {
+  static std::optional<TTo> convert(const std::string& from) {
     if (from.size() == 0) {
       // Return early. Otherwise, empty strings will be parsed as 0, which is
       // misleading.
       return std::nullopt;
     }
 
-    // Amazingly, C++ has no* string parsing functions that work with strings
-    // that might not be null-terminated. So we have to copy to a std::string
-    // (which _is_ guaranteed to be null terminated) before parsing.
-    // * except std::from_chars, but compiler/library support for the
-    //   floating-point version of that method is spotty at best.
-    std::string temp(from);
+    if (from.find('-') != std::string::npos) {
+      // The string must be manually checked for a negative sign because for
+      // std::strtoull accepts negative numbers and bitcasts them, which is not
+      // desired!
+      return std::nullopt;
+    }
+
+    errno = 0;
 
     char* pLastUsed;
-    uint64_t parsedValue = std::strtoull(temp.c_str(), &pLastUsed, 10);
-    if (pLastUsed == temp.c_str() + temp.size()) {
-      // Successfully parsed the entire string as an integer of this type.
-      return CesiumUtility::losslessNarrow<TTo, uint64_t>(parsedValue);
+    uint64_t parsedValue = std::strtoull(from.c_str(), &pLastUsed, 10);
+    if (errno != EINVAL && errno != ERANGE) {
```

Comment:
From browsing the documentation on `strtoull`, I do see `ERANGE` as one of the error values. I don't see `EINVAL` mentioned.

Also, I see that if no conversion can be performed, `0` or `ULONG_MAX or ULLONG_MAX` could be returned too.

Reply:
> I don't see EINVAL mentioned.

Some websites mention that these functions return `EINVAL` if no conversion could be performed (i.e., string did not contain any number). [This one for example](https://man7.org/linux/man-pages/man3/strtol.3.html). I wasn't sure if the functions still had this kind of behavior, so I erred on the side of caution. 

On second glance, most of the sites that mention `EINVAL` look to be written for C and not C++, and it says that `EINVAL` is not used in C99. I don't really know what that means relative to C++, but it seems like the pointer check (`pLastUsed`) is probably sufficient for checking if the entire string was numerical. So I'll remove the references to `EINVAL`.

> Also, I see that if no conversion can be performed, 0 or ULONG_MAX or ULLONG_MAX could be returned too.

I avoid checking for those values because they are valid values that could be read from a valid string. At the end of the day, either `pLastUsed` or `errno` will tell us if something went wrong. I believe that will be sufficient.

Reply:
(not used in C99) <-- Yeah, that's a little ambiguous. Does it mean it's older and stopped being used at C99 and up? Or just C99? I don't know.

Either way, this is an interesting point related to what C++ reference documentation we use. I'd lean Microsoft's [doc](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/strtoull-strtoull-l-wcstoull-wcstoull-l?view=msvc-170#return-value) (updated 2/16/2023), since we develop daily on it. On the other hand, it also implements functions outside of the standard, and since we build on Linux too, and generally don't use those in our platform independent code.

And that doc somewhat ambiguously states that 0 can be returned if no conversation can be performed. In this case would ERANGE get set? Would it still move the pointer in this case? (again, don't know).

I'll leave these thoughts with you.